build/scripts/PullRequestValidation/ValidateIssuesForPullRequest.ps1

```diff
@@ -5,51 +5,100 @@ param(
+    .Description
+    Validates all issues linked to a pull request. All linked issues must be open and approved.
+#>
+function Test-GitHubIssue() {
+    param(
+        [Parameter(Mandatory = $false)]
+        [string] $Repository,
+        [Parameter(Mandatory = $false)]
+        [string[]] $IssueIds,
+        [Parameter(Mandatory = $false)]
+        [object] $PullRequest
+    )
+    $invalidIssues = @()
 
-$Comment = "Could not find linked issues in the pull request description. Please make sure the pull request description contains a line that contains '$issueSection' followed by the issue number being fixed. Use that pattern for every issue you want to link."
-if(-not $issueIds) {
-    $pullRequest.AddComment($Comment)
-    throw $Comment
+    foreach ($issueId in $IssueIds) {
+        Write-Host "Validating issue $issueId"
+        $issue = [GitHubIssue]::Get($issueId, $Repository)
+    
+        # If the issue is not approved, add a comment to the pull request and throw an error
+        $isValid = $issue -and $issue.IsApproved() -and $issue.IsOpen() -and (-not $issue.IsPullRequest())
+        $Comment = "Issue #$($issueId) is not valid. Please make sure you link an **issue** that exists, is **open** and is **approved**."
+        if (-not $isValid) {
+            $PullRequest.AddComment($Comment)
+            $invalidIssues += $issueId
+        }
+        else {
+            $PullRequest.RemoveComment($Comment)
+        }
+    }
+    
+    if($invalidIssues) {
+        throw "The following issues are not open or approved: $($invalidIssues -join ', ')"
+    }
 }
 
-$pullRequest.RemoveComment($Comment)
-
-$invalidIssues = @()
+Write-Host "Validating PR $PullRequestNumber"
 
-foreach ($issueId in $issueIds) {
-    Write-Host "Validating issue $issueId"
-    $issue = [GitHubIssue]::Get($issueId, $Repository)
+$pullRequest = [GitHubPullRequest]::Get($PullRequestNumber, $Repository)
+$issueIds = $pullRequest.GetLinkedIssueIDs()
+$adoWorkitems = $pullRequest.GetLinkedADOWorkitems()
 
-    # Add a comment to the PR for each invalid issue.
-    $isValid = $issue -and $issue.IsApproved() -and $issue.IsOpen() -and (-not $issue.IsPullRequest())
-    $Comment = "Issue #$($issueId) is not valid. Please make sure you link an **issue** that exists, is **open** and is **approved**."
-    if (-not $isValid) {
-        $pullRequest.AddComment($Comment)
-        $invalidIssues += $issueId
-    }
-    else {
-        $pullRequest.RemoveComment($Comment)
-    }
-}
+Test-WorkitemIsLinked -IssueIds $issueIds -ADOWorkItems $adoWorkitems -PullRequest $PullRequest -FromFork $FromFork
 
-if($invalidIssues) {
-    throw "The following issues are not valid: $($invalidIssues -join ', '). Issues linked in a PR need to exist, be open and approved."
+# If the pull request is from a fork, validate the linked issues
+if ($FromFork) {
+    Test-GitHubIssue -Repository $Repository -IssueIds $issueIds -PullRequest $PullRequest
```

Comment:
So, for internal PRs, there will be no validation on the linked issue? The 'approved' check might not be necessary, but what about the other checks: if it's open and not a PR? 🤔

Reply:
Resolved.